packages/kit/src/runtime/server/page/actions.js

```diff
@@ -0,0 +1,179 @@
+import { error, json } from '../../../exports/index.js';
+import { normalize_error } from '../../../utils/error.js';
+import { negotiate } from '../../../utils/http.js';
+import { HttpError, Redirect, ValidationError } from '../../control.js';
+import { error_to_pojo } from '../utils.js';
+
+// Info: `enhance` action and `updateForm` live in `runtime/app/forms.js`
+
+/** @param {import('types').RequestEvent} event */
+export function is_action_json_request(event) {
+	const accept = negotiate(event.request.headers.get('accept') || 'text/html', [
+		'text/html',
+		'application/json'
+	]);
```

Comment:
this should probably default to JSON, since `fetch` has `*/*` and a document request will always favour HTML

```suggestion
	const accept = negotiate(event.request.headers.get('accept') || '*/*', [
		'application/json',
		'text/html'
	]);
```